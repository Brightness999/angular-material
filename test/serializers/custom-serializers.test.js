/**
 * @fileoverview Test custom serializers.
 */
const Logality = require('../../app/logality');
const { stubLogality } = require('../lib/tester.lib');

const DATA = {
  id: 12,
  property: 'zero',
  value: 0.23,
};

describe('Custom Serializers', () => {
  stubLogality();

  test('The build-in custom serializer', () => {
    let outputDone = false;
    const logality = new Logality({
      appName: 'testLogality',
      output: (logMessage) => {
        expect(logMessage).toBeObject();
        expect(logMessage).toMatchSnapshot();
        outputDone = true;
      },
    });

    const log = logality.get();

    log('info', 'hello world', { custom: { a: 1, b: 2 } });
    expect(outputDone).toBeTrue();
  });

  test('Will log custom serializer on context', () => {
    const serializers = {
      zit: (data) => {
        return {
          path: 'context.zit',
          value: data,
        };
      },
    };

    let outputDone = false;
    const logality = new Logality({
      appName: 'testLogality',
      serializers,
      output: (logMessage) => {
        expect(logMessage).toMatchSnapshot();
        outputDone = true;
      },
    });

    const log = logality.get();

    log('info', 'hello world', { zit: DATA });
    expect(outputDone).toBeTrue();
  });

  test('Will log custom serializer on root', () => {
    const serializers = {
      zit: (data) => {
        return {
          path: 'zit',
          value: data,
        };
      },
    };

    let outputDone = false;
    const logality = new Logality({
      appName: 'testLogality',
      serializers,
      output: (logMessage) => {
        expect(logMessage).toMatchSnapshot();
        outputDone = true;
      },
    });

    const log = logality.get();

    log('info', 'hello world', { zit: DATA });
    expect(outputDone).toBeTrue();
  });
});
