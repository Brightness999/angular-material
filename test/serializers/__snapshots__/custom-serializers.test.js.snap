// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom Serializers Will log custom serializer on context 1`] = `
Object {
  "context": Object {
    "runtime": Object {
      "application": "testLogality",
    },
    "source": Object {
      "file_name": "/test/serializers/custom-serializers.test.js",
    },
    "system": Object {
      "hostname": "localhost",
      "pid": 36255,
      "process_name": "node .",
    },
    "zit": Object {
      "id": 12,
      "property": "zero",
      "value": 0.23,
    },
  },
  "dt": "2018-05-18T16:25:57.815Z",
  "event": Object {},
  "level": "info",
  "message": "hello world",
  "severity": 6,
}
`;

exports[`Custom Serializers Will log custom serializer on root 1`] = `
Object {
  "context": Object {
    "runtime": Object {
      "application": "testLogality",
    },
    "source": Object {
      "file_name": "/test/serializers/custom-serializers.test.js",
    },
    "system": Object {
      "hostname": "localhost",
      "pid": 36255,
      "process_name": "node .",
    },
  },
  "dt": "2018-05-18T16:25:57.815Z",
  "event": Object {},
  "level": "info",
  "message": "hello world",
  "severity": 6,
  "zit": Object {
    "id": 12,
    "property": "zero",
    "value": 0.23,
  },
}
`;
